### PR TITLE
[TIMOB-23459]: Null check in getFile()

### PR DESCRIPTION
**Pull request by hieupham007**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-23459

### CODE REVIEW

File: android/modules/filesystem/src/java/ti/modules/titanium/filesystem/FilesystemModule.java [hunk 1]
```diff
@@ -81,6 +81,10 @@ public boolean isExternalStoragePresent()
 	@Kroll.method
 	public FileProxy getFile(KrollInvocation invocation, Object[] parts)
 	{
+		//If directory doesn't exist, return
+		if (parts[0] == null) {
```
Comment (sgtcoolguy) [inline android/modules/filesystem/src/java/ti/modules/titanium/filesystem/FilesystemModule.java]: What if any of the parts are null? Does it cause an NPE or end up with a nonsense "null" string as a path segment?

**Comment by ashcoding** [comment]
Functionally tested this. Crash does not occur any more.
Will wait for comment by @sgtcoolguy to be addressed first before further actions.

File: android/modules/filesystem/src/java/ti/modules/titanium/filesystem/FilesystemModule.java (hunk 1 above)
Comment (hansemannn) [inline android/modules/filesystem/src/java/ti/modules/titanium/filesystem/FilesystemModule.java]: Just out of curiosity: Maybe `Arrays.asList(parts).subList(0, parts.length).contains(null)` or a manual loop?

Comment (sgtcoolguy) [inline android/modules/filesystem/src/java/ti/modules/titanium/filesystem/FilesystemModule.java]: Also, I'd say we should log a warning or error to the user to let them know they passed in a null argument (maybe spit out the args received?) and we're returning null. Or throw a JS Error or something so they can pinpoint the bad call and know why they get back a null reference (or get the error thrown).

Comment (hieupham007) [inline android/modules/filesystem/src/java/ti/modules/titanium/filesystem/FilesystemModule.java]: If the file name is null, it won't make it to Java. I think the check should be efficient

**Comment by ashcoding** [comment]
I'm going to merge and will have another PR to log out a warning.

**Pull request merged**